src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -110,6 +113,93 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
 
         }
 
+        protected void HandleTextInput(string? text, TreeDataGrid treeDataGrid, int selectedRowIndex)
+        {
+            if (text != null && treeDataGrid.Columns != null)
+            {
+                var typedChar = text.ToUpper()[0];
+
+                int now = Environment.TickCount;
+                int time = 0;
+                if (_lastCharPressedTime > 0)
+                {
+                    time = now - _lastCharPressedTime;
+                }
+
+                string candidatePattern;
+                if (time < 500)
+                {
+                    if (_typedWord.Length == 1 && typedChar == _typedWord[0])
+                    {
+                        candidatePattern = _typedWord;
+                    }
+                    else
+                    {
+                        candidatePattern = _typedWord + typedChar;
+                    }
+                }
+                else
+                {
+                    candidatePattern = typedChar.ToString();
+                }
+                foreach (var column in treeDataGrid.Columns)
```

Comment:
What will happen here if multiple rows are marked as searchable? Should we exit this loop as soon as a match is found, or even after the first searchable column?

Reply:
We need validation I think so I am not sure what to do here. Would be better if we would throw exceptions when the user sets IsTextSearchEnabled  for more than one column

Reply:
Ah yes, I forgot about that. I'm fine with leaving it as-is for the moment and adding a check, or support for multiple columns at a later date, just wanted to check if you'd considered it.